From: Eric Biggers <ebiggers@kernel.org>
Subject: kernel/crash: remove inclusion of crypto/sha1.h
Date: Sat, 14 Mar 2026 13:42:43 -0700

Several files related to kernel crash dumps include crypto/sha1.h but
never use any of its functionality.  Remove these includes so that these
files don't unnecessarily come up in searches for which kernel code is
still using the obsolete SHA-1 algorithm.

Link: https://lkml.kernel.org/r/20260314204243.45001-1-ebiggers@kernel.org
Signed-off-by: Eric Biggers <ebiggers@kernel.org>
Cc: Baoquan He <bhe@redhat.com>
Cc: Dave Young <dyoung@redhat.com>
Cc: Vivek Goyal <vgoyal@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 kernel/crash_core.c    |    2 --
 kernel/crash_reserve.c |    2 --
 kernel/vmcore_info.c   |    2 --
 3 files changed, 6 deletions(-)

--- a/kernel/crash_core.c~kernel-crash-remove-inclusion-of-crypto-sha1h
+++ a/kernel/crash_core.c
@@ -27,8 +27,6 @@
 #include <asm/page.h>
 #include <asm/sections.h>
 
-#include <crypto/sha1.h>
-
 #include "kallsyms_internal.h"
 #include "kexec_internal.h"
 
--- a/kernel/crash_reserve.c~kernel-crash-remove-inclusion-of-crypto-sha1h
+++ a/kernel/crash_reserve.c
@@ -20,8 +20,6 @@
 #include <asm/page.h>
 #include <asm/sections.h>
 
-#include <crypto/sha1.h>
-
 #include "kallsyms_internal.h"
 #include "kexec_internal.h"
 
--- a/kernel/vmcore_info.c~kernel-crash-remove-inclusion-of-crypto-sha1h
+++ a/kernel/vmcore_info.c
@@ -18,8 +18,6 @@
 #include <asm/page.h>
 #include <asm/sections.h>
 
-#include <crypto/sha1.h>
-
 #include "kallsyms_internal.h"
 #include "kexec_internal.h"
 
_
